### DISCUSSION TITLE
Unable to connect to Elasticsearch

### DISCUSSION THREAD

**Discussion by MarioBabic** [Security-Onion-Solutions/securityonion#8874] [Unsupported Versions]
I want to connect to elasticsearch remotely but even I can't reach the elasticsearch running on localhost:9200 via curl on the host.
Output:

curl: (52) Empty reply from server

**Comment by reyesj2** [comment]
For a local query you can run
```
so-elasticsearch-query
```
This will return an empty {}

Try this to pull data
```
so-elasticsearch-query '*:so-*/_search' -d '{"query": {"match_all": {}}, "size": 1}' | jq
```

**Comment by MarioBabic** [comment]
It's the same output: curl: (52) Empty reply from server.

**Comment by dougburks** [comment]
Is Elasticsearch working correctly otherwise? Can you query it using SOC Dashboards?

What is the output of the following?
```
sudo so-elasticsearch-query /
```

**Reply by MarioBabic** [reply]
Yes, otherwise Elasticsearch working correctly.

This is output:

{
  "name" : "forenzika",
  "cluster_name" : "forenzika",
  "cluster_uuid" : "l2sngVOAQJ6poinnvtk_VQ",
  "version" : {
    "number" : "8.4.1",
    "build_flavor" : "default",
    "build_type" : "docker",
    "build_hash" : "2bd229c8e56650b42e40992322a76e7914258f0c",
    "build_date" : "2022-08-26T12:11:43.232597118Z",
    "build_snapshot" : false,
    "lucene_version" : "9.3.0",
    "minimum_wire_compatibility_version" : "7.17.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "You Know, for Search"
}

**Reply by dougburks** [reply]
Since that works, you should then be able to query Elasticsearch correctly. For example, to query Elasticsearch for a list of indices:
```
sudo so-elasticsearch-query _cat/indices
```

**Comment by MarioBabic** [comment]
The problem is that I cannot access elasticsearch remote via port 9200 or someone else. I made changes in elasticsearch.yml but it has no effect. What changes should I make in the configuration to enable remote connection to elasticsearch?

**Reply by dougburks** [reply]
Have you allowed the connection through the firewall?
https://docs.securityonion.net/en/2.3/so-allow.html

**Comment by MarioBabic** [comment]
The host and the remote computer I'm trying to access from are on the same network. I tried to make changes in elasticsearch.yml but it didn't work.

**Reply by dougburks** [reply]
Even if they are on the same network, Security Onion has a host-based firewall. You will need to allow the traffic using so-allow as I mentioned in my previous reply. You shouldn't need to modify elasticsearch.yml.
